### ISSUE TITLE
Realm: support for nested OpenMP parallelized loops with collapse

### ISSUE THREAD

**Issue by mariodirenzo** [StanfordLegion/legion#840]
A legion task containing a loop that looks like this 
```
#ifdef REALM_USE_OPENMP
   #pragma omp parallel for collapse(2)
#endif
   for (int k = r_ModCells.lo.z; k <= r_ModCells.hi.z; k++)
      for (int j = r_ModCells.lo.y; i <= r_ModCells.hi.y; j++) {
         double DiffFluxM[5];
         double DiffFluxP[5];

         for (unsigned i=0; i<5; i++)
                DiffFluxM[i] = 1.0; 
         
         // Now loop along the x line
         for (int i = r_ModCells.lo.x; i <= r_ModCells.hi.x; i++) {
            const Point<3> p = Point<3>{i,j,k};

            for (unsigned i=0; i<5; i++)
                DiffFluxM[i] =i;

            // Update time derivative
            for (unsigned i=0; i<5; i++)
               acc_Conserved_t.ptr(p)[i] += (DiffFluxP[i] - DiffFluxM[i]);

            // Store plus flux for next point
            for (unsigned i=0; i<5; i++)
               DiffFluxM[i] = DiffFluxP[i];

         }
      }
```
fails to link in the executable with the error
```
/home/mariodr/Codes/prometeo/src/prometeo_rhs.cc:1047: undefined reference to `GOMP_loop_static_start'
/home/mariodr/Codes/prometeo/src/prometeo_rhs.cc:1047: undefined reference to `GOMP_loop_end_nowait'
/home/mariodr/Codes/prometeo/src/prometeo_rhs.cc:1045: undefined reference to `GOMP_loop_static_next'
```

**Comment by streichler** [comment]
Support for this (and a few other useful things like dynamic schedules and critical sections) are merged into master with commits f2dc82f and 253e52e.

@mariodirenzo can you double check that things still work after the second set of changes as well?

**Comment by mariodirenzo** [comment]
yes, it still works for me
